src/core/Akka.Cluster/Member.cs

```diff
@@ -92,6 +93,20 @@ internal Member(UniqueAddress uniqueAddress, int upNumber, MemberStatus status,
             Roles = roles;
         }
 
+        /// <summary>
+        /// Used when `akka.actor.serialize-messages = on`.
+        /// </summary>
+        /// <param name="uniqueAddress">The address of the member.</param>
+        /// <param name="upNumber">The upNumber of the member, as assigned by the leader at the time the node joined the cluster.</param>
+        /// <param name="status">The status of this member.</param>
+        /// <param name="roles">The roles for this member. Can be empty.</param>
+        [JsonConstructor]
```

Comment:
isn't this serialized via protobuff ClusterMessageSerializer? Why json serialization?

Reply:
@zbynek001 A good question to ask. You're correct that there are a lot of protobuf encoded Cluster Messages (in ClusterMessages.g.cs), however this particular Member.cs class is actually being referenced in the messages defined under ClusterEvent.XYZ which are messages published to the EventBus (and in the context of the BugFix3724Spec test at least, are destined for the Event.LoggingBus).

I am new to the akka.net code base so take the following comments as very tentative:

- The messages under ClusterEvent.XYZ are JSON serialized by design at this point
- My gut says that protobuf would be better (even if we are waiting for [Hyperion](https://github.com/akkadotnet/akka.net/pull/3725#discussion_r260521727)) but that would be a breaking change
- I'm keen to circle back to [this comment](https://github.com/akkadotnet/akka.net/pull/3725#discussion_r260521054) in the context of netstandard2.0 to see whether #3790 will enable the INoSerializationVerificationNeeded restriction to be lifted

@Aaronontheweb Do you have any input on this point?

Reply:
So after a quick tour through the code, I think that the underlying Gossip and state management messages for the cluster do use the protobuf messages, but the observable messages published by the ClusterDomainEventPublisher on the EventBus are JSON serialized.

I'm interested to know if this is a specific design decision and/or if there would be resistance to migrating to a consistent protobuf wire format for both?

Reply:
@zbynek001 @heatonmatthew so this might be an issue with `serialize-messages` inside Akka core where we don't use the correct serializer based on its manifest type.... I'm going to see if I can fix that in https://github.com/akkadotnet/akka.net/pull/3744 - I added a `JsonConstructor` call similar to this one for the `Member` class and that may no longer be necessary once the serialization fixes are added to serialize-messages in that PR.

Reply:
@Aaronontheweb let me know if there's anything specific you could do with a hand on. It's quite interesting code to be working on.